src/tools/chip-cert/dacs.py

```diff
@@ -0,0 +1,279 @@
+
+    def generic_arrays(self) -> tuple[str, str]:
+        """Returns the generic refernce for Certs and Keys of this type"""
+        h_start = 'extern ByteSpan ' + self.own.generic_prefix
+        h = h_start + 'Cert;\n' + h_start + 'PublicKey;\n' + h_start + 'PrivateKey;\n\n'
+        c_start = 'ByteSpan ' + self.own.generic_prefix
+        cert = c_start + \
+            'Cert       = ByteSpan(' + self.own.cert_array_name + ');\n'
+        priv = c_start + \
+            'PrivateKey = ByteSpan(' + self.own.private_key_array_name + ');\n'
+        publ = c_start + \
+            'PublicKey  = ByteSpan(' + self.own.public_key_array_name + ');\n'
+        c = cert + priv + publ
+        return [h, c]
+
+    def headers(self, filename) -> tuple[str, str]:
+        """Returns a string with the headers for the output files"""
+        includes = '#include <cstdint>\n\n#include <lib/support/Span.h>\n#include <platform/CHIPDeviceConfig.h>\n\n'
+        namespaces = 'namespace chip {\nnamespace DevelopmentCerts {\n\n'
+        h_top = copyrightNotice + '#pragma once\n' + includes + namespaces
+
+        c_top = copyrightNotice + '#include "' + \
+            filename + '.h"\n\n' + includes + namespaces
+        return [h_top, c_top]
+
+    def footer(self) -> str:
+        """Returns the footer for the output files"""
+        return '} // namespace DevelopmentCerts\n} // namespace chip\n'
+
+
+def main():
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument('-d', '--dev_dir', dest='certdir',
+                           default='credentials/development/attestation',
+                           help='output directory for PEM and DER files')
+    argparser.add_argument('-t', '--test_dir', dest='testdir',
+                           default='credentials/test/attestation/', help='directory holding test PAA')
+    argparser.add_argument('-o', '--out_dir', dest='outdir',
+                           default='src/credentials/examples/', help='Output directory for array files')
+    argparser.add_argument('-c', '--chip-cert_dir', dest='chipcertdir',
+                           default='out/debug/standalone/', help='Directory where chip-cert tool is located')
+    argparser.add_argument('--paiout', dest='paiout', default='ExamplePAI',
+                           help='output file name for PAI C arrays (no .h or .cpp extension)')
+    argparser.add_argument('--dacout', dest='dacout', default='ExampleDACs',
+                           help='output file name for PAI C arrays (no .h or .cpp extension)')
+
+    args = argparser.parse_args()
+    builder = DevCertBuilder(CertType.PAI, args.certdir,
+                             args.testdir, args.chipcertdir, 0x8000)
+
+    builder.make_certs_and_keys()
+    [h_full, c_full] = builder.full_arrays()
+    [h_generic, c_generic] = builder.generic_arrays()
+    [h_top, c_top] = builder.headers(args.paiout)
+    footer = builder.footer()
+    with open(args.outdir + args.paiout + '.h', "w") as hfile:
```

Comment:
consider using (Jinja 2 Templates)[https://jinja.palletsprojects.com/en/3.0.x/] as we already use them in places and they make for a clearer "what is the generated file content" file.